### ISSUE TITLE
ESP32-CAM

### ISSUE THREAD

**Issue by Androxilogin** [frakman1/ContaCam#18]
Are these unsupported by the program? I've tried all of the generic options and a few others in the list. The cameras themselves give ex: http://192.168.1.202:81/stream for a connection. I saw only one mention of ContaCam as a recommendation for these devices but the thread was closed years ago on some page I landed on. I can stream it from VLC but that seems to be it so far.

**Comment by frakman1** [comment]
I have not heard of this device but if it supports RTSP streams, then you should be able to add it in ContaCam. 
Have you looked at [this](https://github.com/rzeldent/esp32cam-rtsp). It looks like it would give you an RTSP feed that you can use with Contacam.

If not, you may want to try [go2rtc](https://github.com/AlexxIT/go2rtc) to convert your stream into something suitable for ContaCam.

If you know of, or can point to a public ESP32-CAM stream, I can try playing around with it.

**Comment by Androxilogin** [comment]
These are both interesting options. I decided to try out go2rtc first since the two cams I have were already running and connected to the wifi out in the garage and it worked. 

The other option seems more interesting since it allows you to connect to it as a hotspot to set up credentials which would be useful when switching between networks so I'll try flashing that as well. 

To be honest with you, I didn't know the difference between motion jpeg and rtsp or if it was capable of either. I assume jpeg takes stills frame by frame while rtsp provides a constant stream? Either way, I guess the stream is neither but this lightweight server program did the job. Thanks for the info!

**Comment by frakman1** [comment]
Glad I could help. I personally use `go2rtc` for converting various streams I use in and out of my home network. I love how it has a whole page dedicated to the various links and formats that it can provide. I can always find one that works.

RTSP is really the way to go as it is universally supported. MJPEG is as you described but does not support audio and I think is slower. Good luck.

**Comment by Androxilogin** [comment]
A little update: 

I tried the first option you suggested and it seems to be working great as well. A version of the code [**I found previously**](https://github.com/s60sc/ESP32-CAM_MJPEG2SD) offered more options but required an SD card for settings. Most of the options I would never use either and I couldn't get it to interface with ContaCam anyway so discarded it. 

The quality of these things is pretty awful at night but I just wanted something to see when the mail carrier came so it's nice to have found a $5 solution for that. Thanks again!
